### PR TITLE
chore: Removing SentryFrameTracker atomic operation

### PR DESCRIPTION
**Pull request by brustolin**
SentryFrameTracker makes uses of CADisplayLink to increment the number of slow and frozen frames.
CADisplayLink only uses the main thread, we dont need to worry about a variable increment discrepancy. 

This also removes the slowest Test in our CI, responsible for almost 15% of unit test time alone.

_#skip-changelog_

### CODE REVIEW

**Review by philipphofmann — COMMENTED** [review]
I think it's not a good idea to remove the test.

File: Tests/SentryTests/Integrations/Performance/FramesTracking/SentryFramesTrackerTests.swift [hunk 1]
```diff
@@ -87,36 +87,7 @@ class SentryFramesTrackerTests: XCTestCase {
 
         try assert(slow: 1, frozen: 1, total: 2, frameRates: 2)
     }
-    
-    func testAllFrames_ConcurrentRead() throws {
```
[the remark this answers is not in the document]
Comment (brustolin) [reply Tests/SentryTests/Integrations/Performance/FramesTracking/SentryFramesTrackerTests.swift:91]: What do you think this test would show us?
What would you expect to be the failing condition?

[the remark this answers is not in the document]
Comment (brustolin) [reply Tests/SentryTests/Integrations/Performance/FramesTracking/SentryFramesTrackerTests.swift:91]: For this feature specifically I don't see any value testing for thread safety.
We are incrementing and reading to and from an unsigned integer.
This is the type of behavior that don't crash with multiple threads.

We could argue that the data may not be precise between writing and reading, but this was also true before, when we were using atomic operations, and something we cant really control.

File: Sources/Sentry/SentryFramesTracker.m [hunk 2]
```diff
@@ -215,9 +203,9 @@ - (void)recordTimestamp:(uint64_t)timestamp value:(NSNumber *)value array:(NSMut
 
 - (SentryScreenFrames *)currentFrames
 {
-    NSUInteger total = atomic_load_explicit(&_totalFrames, SentryFramesMemoryOrder);
-    NSUInteger slow = atomic_load_explicit(&_slowFrames, SentryFramesMemoryOrder);
-    NSUInteger frozen = atomic_load_explicit(&_frozenFrames, SentryFramesMemoryOrder);
+    NSUInteger total = _totalFrames;
+    NSUInteger slow = _slowFrames;
+    NSUInteger frozen = _frozenFrames;
```
Comment (philipphofmann) [inline Sources/Sentry/SentryFramesTracker.m:208]: `l`: I think we could directly pass these values to the init of `SentryScreenFrames`. No need to create local variables.

File: Tests/SentryTests/Integrations/Performance/FramesTracking/SentryFramesTrackerTests.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (philipphofmann) [reply Tests/SentryTests/Integrations/Performance/FramesTracking/SentryFramesTrackerTests.swift:91]: Yep, that's true; not worth it now anymore. I thought there could me some other synchronization things that could break in the code, but there aren't any.

**Review by philipphofmann — APPROVED** [review]
Thanks @brustolin 👍

**Review by armcknight — APPROVED** [review]
Nice work @brustolin, thanks!

**Pull request merged**